lib/src/main/java/com/sensirion/libble/devices/BleDevice.java

```diff
@@ -164,4 +177,25 @@
      */
     @SuppressWarnings("unused")
     boolean areAllServicesReady();
+
+    /**
+     * Comparator for comparing the different {@link BleService} using their
+     * {@link BleServiceSynchronizationPriority} for comparing them.
+     */
+    @NonNull
+    Comparator<BleService> SERVICE_PRIORITY_COMPARATOR = new Comparator<BleService>() {
+        @Override
+        public int compare(@NonNull final BleService serviceA,
+                           @NonNull final BleService serviceB) {
+            switch (serviceA.getServiceSynchronizationPriority()) {
```

Comment:
that whole method looks a lot more complicated than it should be:
return serviceB.getServiceSynchronizationPriority().ordinal() - serviceA.getServiceSynchronizationPriority().ordinal()